packages/hash/api/src/graph/index.ts

```diff
@@ -51,6 +51,29 @@ const httpsAgent = new HttpsAgent(agentConfig);
 
 export type GraphApi = GraphApiClient & DataSource;
 
+const isErrorAxiosError = (error: Error): error is AxiosError =>
+  (error as AxiosError).isAxiosError;
+
+export class GraphApiError extends Error {
+  // `status` is the HTTP status code from the server response
+  status?: number;
+  // `payload` is the contents of the HTTP body from the server response
+  payload: any;
```

Comment:
I can constrain it to be a JSON object, because I know that's the only thing that can be returned in the HTTP body, but it'd be nice to constrain it further.

@hashintel/backend are there any additional assumptions I can make about the error response of the Graph API? In most of the errors I encountered whilst testing the response body was a `string`, is that something I can assume will always be the case?

Reply:
cc @TimDiekmann

Reply:
I probably wouldn't constrain it to be a JSON object because I'm not sure if we will stick with JSON forever. Is there a specific issue keeping this as an `any`?

Reply:
The issue with `any` is that it offers no type safety or information about what we expect the shape of the response data to be. If it is guaranteed JSON now I would type it as such, and we can update it when we change it.

Reply:
I've constrained the payload to JSON in https://github.com/hashintel/hash/pull/1702/commits/84d133c21cf46c424b6ad624ff6615ac050f487a, but @hashintel/backend I think we should strongly consider having a consistent approach in the Graph API to what is returned in this HTTP body. Even if it's just something simple like:
```JSON
{
  "code": "ERROR_CODE",
  "message": "The error message.",
}
```